helix_fhir_client_sdk/utilities/retryable_aiohttp_client.py

```diff
@@ -90,12 +91,17 @@ async def fetch(
         while retry_attempts < self.retries:
             retry_attempts += 1
             try:
-                if self.chunked:
-                    kwargs["chunked"] = self.chunked
-                if self.compress:
-                    kwargs["compress"] = self.compress
                 if headers:
                     kwargs["headers"] = headers
+                # if there is no data then remove from kwargs so as not to confuse aiohttp
+                if "data" in kwargs and kwargs["data"] is None:
+                    del kwargs["data"]
+                # compression and chunked can only be enabled if there is content sent
+                if "data" in kwargs and kwargs["data"] is not None:
+                    if self.send_data_as_chunked:
+                        kwargs["chunked"] = self.send_data_as_chunked
+                    if self.compress:
+                        kwargs["compress"] = self.compress
```

Comment:
If data is present and send_data_as_chunked is set to False, Aiohttp would still set it to True based on the value of compress. This will set Transfer-Encoding header that could yield unexpected result.

Reply:
Check self.update_content_encoding(data) and self.update_transfer_encoding() calls in aiohttp/client_reqrep.py::ClientRequest.__init__
update_content_encoding --> sets self.chunked to True based on self.compress 
update_transfer_encoding --> sets Transfer-Encoding header based on self.chunked.